### PR TITLE
feature: make 'pouch ps' show running time and add flags in ps command.

### PR DESCRIPTION
**Pull request by zeppp**
Signed-off-by: zeppp <zeppp1995@gmail.com>

**1.Describe what this PR did**
1. make 'pouch ps' show running time.
2. fix a bug that the 'Created' shows wrong time when start a container.
3. add flags "quiet" and "all" in ps command because **now `pouch ps` only shows running and paused containers by default.**
4. refactor FormatTimeInterval fuction in utils.go.

**2.Does this pull request fix one issue?** 
NONE. It is a follow-up PR of #358.

**3.Describe how you did it**

**4.Describe how to verify it**
```
→ pouch ps
Name   ID       Status          Created          Image                              Runtime
2      e42c68   Up 26 minutes   27 minutes ago   docker.io/library/busybox:latest   runc
1      a8c2ea   Up 27 minutes   28 minutes ago   docker.io/library/busybox:latest   runc

→ pouch ps -a
Name   ID       Status          Created          Image                              Runtime
3      faf132   created         11 minutes ago   docker.io/library/busybox:latest   runc
2      e42c68   Up 27 minutes   27 minutes ago   docker.io/library/busybox:latest   runc
1      a8c2ea   Up 28 minutes   28 minutes ago   docker.io/library/busybox:latest   runc

→ pouch pause 1
→ pouch ps 
Name   ID       Status                  Created          Image                              Runtime
2      e42c68   Up 38 minutes           39 minutes ago   docker.io/library/busybox:latest   runc
1      a8c2ea   Up 39 minutes(paused)   40 minutes ago   docker.io/library/busybox:latest   runc

→ pouch ps -q
e42c68
a8c2ea

→ pouch ps -q -a
faf132
e42c68
a8c2ea

→ pouch unpause 1
→ pouch stop 1
→ pouch start 1
→ pouch ps
Name   ID       Status          Created          Image                              Runtime
2      e42c68   Up 41 minutes   42 minutes ago   docker.io/library/busybox:latest   runc
1      a8c2ea   Up 8 seconds    43 minutes ago   docker.io/library/busybox:latest   runc


```
**5.Special notes for reviews**
**Please remove all containers first and then make install to apply this PR.**

### CODE REVIEW

File: apis/swagger.yml [hunk 1]
```diff
@@ -1186,9 +1186,7 @@ definitions:
         additionalProperties:
           type: "string"
       State:
-        type: "string"
```
Comment (allencloud) [inline apis/swagger.yml]: We need to be very careful to edit swagger.yml which defines API. I agree with your ideas totally.
From my point of view, this pr added feature in cli side. The first question we have is that can we use the current API to make it. If not, we can make API daemon side provide more data in the current API. If still not, we can change API at that time.

In this PR, change on type of `State` is not what we want if we can find other solutions.

File: cli/ps.go [hunk 2]
```diff
@@ -40,7 +44,9 @@ func (p *PsCommand) Init(c *Cli) {
 
 // addFlags adds flags for specific command.
 func (p *PsCommand) addFlags() {
-	// TODO: add flags here
+	flagSet := p.cmd.Flags()
+	flagSet.BoolVarP(&p.flagAll, "all", "a", false, "Show all containers (default shows just running)")
```
Comment (allencloud) [inline cli/ps.go]: We should add this parameter support in daemon side rather than only in client side. WDYT?

File: cli/ps.go [hunk 3]
```diff
@@ -55,25 +61,71 @@ func (p *PsCommand) runPs(args []string) error {
...
+
+$ pouch ps -q
+e42c68
+a8c2ea
+
+$ pouch ps -a -q
+faf132
+e42c68
+a8c2ea
+
```
Comment (allencloud) [inline cli/ps.go]: please remove this redundant blank line 😄

**Comment by zeppp** [comment]
PTAL @allencloud

**Comment by zeppp** [comment]
Updated the fix of #507 into this PR.
PTAL @allencloud

File: apis/server/container_bridge.go [hunk 4]
```diff
@@ -216,28 +218,48 @@ func (s *Server) attachContainer(ctx context.Context, rw http.ResponseWriter, re
 }
 
 func (s *Server) getContainers(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+	option := &mgr.ContainerListOption{
+		All: httputils.BoolValue(req, "all"),
```
Comment (allencloud) [inline apis/server/container_bridge.go]: If we support a flag in daemon side, I think it is changing API, then we should consider updating swagger.yml in parameter part like:
```
        - name: "all"
          in: "query"
          description: "Return all containers. By default, only running containers are shown"
          type: "boolean"
          default: false
```
In API endpoint `GET /containers/json` in swagger.yml.

File: apis/server/container_bridge.go [hunk 5]
```diff
@@ -216,28 +218,48 @@ func (s *Server) attachContainer(ctx context.Context, rw http.ResponseWriter, re
...
-			Status:     string(m.State.Status),
 			Image:      m.Config.Image,
 			Command:    strings.Join(m.Config.Cmd, " "),
-			Created:    m.State.StartedAt,
+			Created:    t.UnixNano(),
 			Labels:     m.Config.Labels,
 			HostConfig: m.HostConfig,
 		}
+
+		if m.State.Status == types.StatusRunning || m.State.Status == types.StatusPaused {
```
Comment (allencloud) [inline apis/server/container_bridge.go]: I think in the future this part of code could be encapsulated into a single function, like `genStatus(State)`.

**Comment by Letty5411** [comment]
@zeppp Could you add a test in test/api_container_list_test.go to test the "all" query param? Also a test in test/cli_ps_test.go to test the "-q" option is needed. Thanks~

File: apis/swagger.yml [hunk 6]
```diff
@@ -1458,9 +1465,12 @@ definitions:
         items:
           $ref: "#/definitions/MountPoint"
       NetworkSettings:
-        additionalProperties:
-          $ref: "#/definitions/EndpointSettings"
-          x-nullable: true
+        type: "object"
+        properties:
+          etworks:
```
Comment (allencloud) [inline apis/swagger.yml]: s/etworks/networks

**Comment by zeppp** [comment]
Updated @Letty5411 @allencloud

**Comment by allencloud** [comment]
LGTM

**Pull request merged**